lib/Echidna/Exec.hs

```diff
@@ -69,25 +70,30 @@ vmExcept e = throwM $ case VMFailure e of {Illegal -> IllegalExec e; _ -> Unknow
 -- using the given execution strategy, handling errors with the given handler.
 execTxWith :: (MonadState x m, Has VM x) => (Error -> m ()) -> m VMResult -> Tx -> m (VMResult, Int)
 execTxWith h m t = do
-  (og :: VM) <- use hasLens
-  setupTx t
-  gasIn <- use $ hasLens . state . gas
-  res <- m
-  gasOut <- use $ hasLens . state . gas
-  cd  <- use $ hasLens . state . calldata
-  case (res, t ^. call) of
-    (f@Reversion, _) -> do
-      hasLens .= og
-      hasLens . state . calldata .= cd
-      hasLens . result ?= f
-    (VMFailure x, _) -> h x
-    (VMSuccess (ConcreteBuffer bc), SolCreate _) ->
-      (hasLens %=) . execState $ do
-        env . contracts . at (t ^. dst) . _Just . contractcode .= InitCode ""
-        replaceCodeOfSelf (RuntimeCode bc)
-        loadContract (t ^. dst)
-    _ -> pure ()
-  pure (res, fromIntegral (gasIn - gasOut))
+  sd <- hasSelfdestructed (t ^. dst)
+  if sd 
+  then pure (VMFailure (Revert ""), 0)
+  else do 
+    hasLens . traces .= emptyEvents 
+    (og :: VM) <- use hasLens
+    setupTx t
+    gasIn <- use $ hasLens . state . gas
+    res <- m
+    gasOut <- use $ hasLens . state . gas
+    cd  <- use $ hasLens . state . calldata
```

Comment:
either align all the `<-`s or align none of them

Reply:
Fixed.